src/Standards/Generic/Tests/Arrays/DisallowCompactArrayBuilderUnitTest.1.inc

```diff
@@ -0,0 +1,33 @@
+<?php
+$var = compact();
+$var = array(1,2,3);
+$var = compact('a','b','c');
+echo $var[1];
+$foo = compact($var[1],$var[2]);
+$foo = compact(
+        'a',
+        "b",
+        'c'
+       );
+$var = compact/*comment*/('a', 'b', "c");
+
+$var = compact(['aa', 'bb' => 'cc']);
+
+$var = compact(array('aa', 'bb' => 'cc'));
+
+$ver = $foo->compact('a', 'b');
+
+function foo($compact) {}
+
+$compact = function ($a, $b, $c) use ($foo): array {};
+$compact('a', 'b', 'c');
+
+view('some.view', compact("a", 'b', 'c'));
+view('some.view', compact(
+    'a',
+    'b',
+    'c'
+));
+
+$var = compact('aa', 'invalid-var.name');
+$var = Bazz::compact('a', 'b');
```

Comment:
I see quite some code in this test case file which doesn't seem to be testing anything in the sniff... ?

Here are some more test cases to take into account:
```php
compact(...$names);
compact( 'prefix' . $name, '$name' . 'suffix', "some$name");
compact(...get_names('category1', 'category2'));

// Last test in the file.
// Live coding/parse error.
compact( 'a', 'b'
```

See this as an invitation to get a little more creative with the tests and to make sure the sniff handles those situations correctly.

Reply:
Most of these were already covered as sniff requires only strings and commas within parenthesis. I've added them to test case. Last one was actually causing error so I fixed that as well.